.github/workflows/check.yml

```diff
@@ -0,0 +1,165 @@
+# This workflow performs static analysis and checks coding style
```

Comment:
What do you think about renaming this file to `style.yml`, `style_check.yml` or something else that would indicate that this is about codding style and not actual tests?

Reply:
There are multiple check types.

Reply:
It is a common name i suppose.